crates/protocol/src/channel_out.rs

```diff
@@ -32,10 +31,10 @@ pub enum ChannelOutError {
 }
 
 /// [ChannelOut] constructs a channel from compressed, encoded batch data.
-#[derive(Debug, Clone)]
+#[allow(missing_debug_implementations)]
 pub struct ChannelOut<'a, C>
 where
-    C: Compressor + Clone + core::fmt::Debug,
+    C: ChannelCompressor,
```

Comment:
looks like this can be relaxed

Reply:
wdym, should I add back the `Clone + core::fmt::Debug` restriction?

Reply:
you can get rid of this where clause here entirely because it's not necessary for the struct def itself